### PR TITLE
Adding AutomationML serializer

### PR DESCRIPTION
**Pull request by mjacoby**
This PR adds the AutomationML serializer.

### CODE REVIEW

File: dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/deserialize/AmlDeserializerTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class AmlDeserializerTest {
+
+    private final AmlDeserializer deserializer = new AmlDeserializer();
+
+    @Test
+    @Ignore
```
[the remark this answers is not in the document]
Comment (mjacoby) [reply dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/deserialize/AmlDeserializerTest.java:33]: Yes, as AmlDeserializer is not yet fully implemented and therefore the test would fail.

File: dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/serialize/AmlSerializerTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.adminshell.aas.v3.dataformat.aml.serialize;
+
+import io.adminshell.aas.v3.dataformat.aml.fixtures.FullExample;
```
[the remark this answers is not in the document]
Comment (mjacoby) [reply dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/serialize/AmlSerializerTest.java:18]: Yes, you are right (in general). We are already working on this, however, FullExample is currently not 100% consistent acrosss different serialization formats. The reason for this is, that when implementing AML we  realized that the full example is actually not 100% valid according to the constraints but this has not caused issues in other serialization formats as they are more "error-friendly" e.g. unresolvable references do not cause JSON serializer to crash as references are just serialized as string whereas in AML unresolvable references can cause a crash because sometimes these references are resolved and the referenced element is inlined.
We are working on unifying FullExample.java but we postponed this task until validation is implemeted and we can be sure that we fix all issues in the FullExample in one go as this requires slight adjustments to the expected results of the unit tests of all serializers.

File: dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/serialize/AmlSerializerTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                .idGenerator(new IntegerIdGenerator())
+                .build());
+        System.out.println(actual);
+        Diff diff = DiffBuilder
+                .compare(expected)
+                .withTest(actual)
+                .normalizeWhitespace()
+                .withNodeMatcher(new DefaultNodeMatcher(
+                        ElementSelectors.conditionalBuilder()
+                                .whenElementIsNamed("Attribute")
```
[the remark this answers is not in the document]
Comment (mjacoby) [reply dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/serialize/AmlSerializerTest.java:56]: These are instructions on how to compare the generated XML/AML to the expected XML/AML. The used framework only works on XML and compares nodes of the XML tree based on node name which in case of AML is most of the time either Attribute or InternalElement. As in AML each element typically has multiple sub-elements of type Attribute and/or InternalElement. These statements specify that e.g. an element is named "Attribute" it should be compare not with another element named "Attribute" at the same position in the other file but with another element that is also named "Attribute" and has the same value for attribute "Name". 
Basically, these rules define how to uniquely identify an Attribute/InternalElement/ExternalInterface so that order of elements is not important when comparing AML files.

File: dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/deserialize/AmlDeserializerTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import io.adminshell.aas.v3.dataformat.aml.AmlDeserializer;
+import io.adminshell.aas.v3.dataformat.aml.fixtures.FullExample;
+import io.adminshell.aas.v3.model.AssetAdministrationShellEnvironment;
+import java.io.FileNotFoundException;
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class AmlDeserializerTest {
```
[the remark this answers is not in the document]
Comment (mjacoby) [reply dataformat-aml/src/test/java/io/adminshell/aas/v3/dataformat/aml/deserialize/AmlDeserializerTest.java:28]: I'm being a bit blind...what exactly is wrong with the package? What should be the correct package?
